src/api/data/cache/charthelper/chart_package_helper.go

```diff
@@ -29,11 +29,13 @@ var DownloadAndExtractChartTarball = func(chart *models.ChartPackage) error {
 
 	if !tarballExists(chart) {
 		if err := downloadTarball(chart); err != nil {
+			cleanChartDataDir(chart)
```

Comment:
Just a FYI:

```
var DownloadAndExtractChartTarball = func(chart *models.ChartPackage) (err error) {
 	if err := ensureChartDataDir(chart); err != nil {
 		return err
 	}

 	defer func() {
 	 	if err != nil {
 	 	 	cleanChartDataDir(chart)
 	 	}
 	}()

 	if !tarballExists(chart) {
  		if err := downloadTarball(chart); err != nil {
			return err
  		}
  	}

 	if err := extractFilesFromTarball(chart); err != nil {
  		return err
  	}
  
 	return nil
}
```

this ensures that if you add another place where you return an error you cannot forget to delete the char data dir when the error happens.